Code/Tools/AssetProcessor/Platform/Linux/native/FileWatcher/FileWatcher_linux.cpp

```diff
@@ -20,150 +23,151 @@ static constexpr size_t s_iNotifyMaxEntries = 1024 * 16;         // Control the
 static constexpr size_t s_iNotifyEventSize = sizeof(struct inotify_event);
 static constexpr size_t s_iNotifyReadBufferSize = s_iNotifyMaxEntries * s_iNotifyEventSize;
 
-struct FolderRootWatch::PlatformImplementation
+bool FileWatcher::PlatformImplementation::Initialize()
 {
-    PlatformImplementation() = default;
-
-    int                         m_iNotifyHandle = -1;
-    QMutex                      m_handleToFolderMapLock;
-    QHash<int, QString>         m_handleToFolderMap;
-
-    bool Initialize()
+    if (m_iNotifyHandle < 0)
     {
+        // The CLOEXEC flag prevents the inotify watchers from copying on fork/exec
+        m_iNotifyHandle = inotify_init1(IN_CLOEXEC);
+
         if (m_iNotifyHandle < 0)
         {
             // The CLOEXEC flag prevents the inotify watchers from copying on fork/exec
             m_iNotifyHandle = inotify_init1(IN_CLOEXEC);
+            const auto err = errno;
+
+            AZStd::fixed_string<255> errorString;
+            errorString.resize_no_construct(errorString.capacity());
+            if (m_iNotifyHandle < 0)
+            {
+                AZ_Warning("FileWatcher", false, "Unable to initialize inotify, file monitoring will not be available: %s\n", strerror_r(err, errorString.data(), errorString.size()));
+            }
```

Comment:
The if statement isn't needed in this case
You can pass `m_iNotifyHandle` directly to AZ_Warning.

I also think the variable should be called `m_inotifyHandle` to avoid any confusion with Hungarian notation

```suggestion
            AZ_Warning("FileWatcher", m_iNotifyHandle < 0, "Unable to initialize inotify, file monitoring will not be available: %s\n", strerror_r(err, errorString.data(), errorString.size()));
```